### ISSUE TITLE
Fading in Function::play

### ISSUE THREAD

**Issue by telephon** [supercollider/supercollider#3284]
I just gave an introduction to delays to the beginner students and I found it really awkward that `Function::play` adds a ramp to the beginning of the sound. I know that we've had discussions about this before, and it was then decided that it is more important to avoid clicks when playing back a signal that starts not at zero.

But I really find it more inconvenient or inconsistent having to explain why, instead of

`{ Impulse.ar(0) }.play`

I need to write

`{ Out(0, Impulse.ar(0)) }.play`

than having to explain why, instead of

`{ SinOsc.ar(60, 0.5pi) }.play`

in order to avoid a click, I have to write

`{ SinOac.ar(60, 0.5pi) * Line.ar(0, 1, 0.001) }.play`

The last one is clear and nothing hidden goes on.

Maybe we should make this behavior configurable? I'm curious to hear what you think is best.

**Comment by redFrik** [comment]
i usually say that there's a built in fade in and to get rid of one can do...
{ Impulse.ar(0) }.play(fadeTime:0)
and most are happy with that.
_f

> 9 nov. 2017 kl. 15:08 skrev Julian Rohrhuber <notifications@github.com>:
> 
> I just gave an introduction to delays to the beginner students and I found it really awkward that Function::play adds a ramp to the beginning of the sound. I know that we've had discussions about this before, and it was then decided that it is more important to avoid clicks when playing back a signal that starts not at zero.
> 
> But I really find it more inconvenient or inconsistent having to explain why, instead of
> 
> { Impulse.ar(0) }.play
> 
> I need to write
> 
> { Out(0, Impulse.ar(0)) }.play
> 
> than having to explain why, instead of
> 
> { SinOsc.ar(60, 0.5pi) }.play
> 
> in order to avoid a click, I have to write
> 
> { SinOac.ar(60, 0.5pi) * Line.ar(0, 1, 0.001) }.play
> 
> The last one is clear and nothing hidden goes on.
> 
> Maybe we should make this behavior configurable? I'm curious to hear what you think is best.
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
> 


  #|
     fredrikolofsson.com     musicalfieldsforever.com
  |#

**Comment by telephon** [comment]
hm, ok, that's also a possibility.

**Comment by snappizz** [comment]
i've oscillated on this a few times. the arguments for removing the fade in are strong, but i'm worried about breaking backward compatibility. people can and do use `Function:play` in actual compositions, even if they shouldn't.

**Comment by telephon** [comment]
I do think they should use `Function:play`. It is clear and in many situations very adequate. I'd just find it more adequate if it were not fading in ;). But my suggestion was more like making this a global option.

Not such a big issue, in any case.

**Comment by brianlheim** [comment]
I think that we should keep it as is, since it is easy to configure. Rather than add to global state I would rather just have a second function with fadeTime defaulting to 0. Then the code and documentation are in the same place, and you can use both approaches easily in the same session.

**Comment by vivid-synth** [comment]
Maybe a note in the docs about ```.play(fadeTime:0)```?

**Comment by brianlheim** [comment]
@vivid-synth yeah, it could probably be emphasized a little. right now it's mentioned obliquely - "A Linen is also added to avoid clicks"
